lib/java_buildpack/framework/app_dynamics_agent.rb

```diff
@@ -83,7 +84,7 @@ def host_name(java_opts, credentials)
 
       def node_name(java_opts, credentials)
         name = credentials['node-name'] || @configuration['default_node_name']
-        java_opts.add_system_property('appdynamics.agent.nodeName', "#{name}")
+        java_opts.add_system_property('appdynamics.agent.nodeName', @appdynamics_tier_name + ":#{name}")
```

Comment:
@pranta @nebhale Sorry Let me talk with some of my custom system engineers and see what they think.  I'll get back soon.